src/common/input/config.rs

```diff
@@ -44,6 +47,34 @@ impl Default for Config {
     }
 }
 
+impl TryFrom<&CliArgs> for Config {
+    type Error = ConfigError;
+
+    fn try_from(opts: &CliArgs) -> Result<Self, Self::Error> {
```

Comment:
Does it make sense to use `ConfigResult` here, or do you think that this is clearer?

Reply:
Thank you. It makes sense :+1: